kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt

```diff
@@ -0,0 +1,22 @@
+package io.kotest.matchers.sets
+
+import io.kotest.assertions.print.print
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+
+infix fun <T> Set<T>.shouldIntersect(ts: Set<T>) = this should intersectWith(ts)
+infix fun <T> Set<T>.shouldNotIntersect(ts: Set<T>) = this shouldNot intersectWith(ts)
+
+fun <T> intersectWith(ts: Set<T>): Matcher<Set<T>> = object : Matcher<Set<T>> {
+   override fun test(value: Set<T>): MatcherResult {
+      val intersection = ts.filter { value.contains(it) }
```

Comment:
https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/intersect.html

Can't actually believe this function exists in stdlib

```suggestion
      val intersection = ts intercect value
```